File: ios/MenCare/Features/Staff/AppointmentDetailScreen.swift
import SwiftUI
import Observation

@MainActor
@Observable
final class AppointmentDetailViewModel {

    private let booking: BookingRepository
    private let crm: CrmRepository
    private let catalog: CatalogRepository
    private let appointmentId: String

    let state = LoadState()

    private(set) var appointment: Appointment?
    private(set) var client: ClientRecord?
    private(set) var services: [String: Service] = [:]
    private(set) var history: [Appointment] = []
    var actionError: String?

    init(booking: BookingRepository, crm: CrmRepository, catalog: CatalogRepository, appointmentId: String) {
        self.booking = booking
        self.crm = crm
        self.catalog = catalog
        self.appointmentId = appointmentId
    }

    func load() async {
        await state.run {
            let catalogue = try await catalog.services()
            services = Dictionary(uniqueKeysWithValues: catalogue.map { ($0.id, $0) })

            let found = try await booking.appointment(appointmentId)
            appointment = found
            guard let found else {
                client = nil
                history = []
                return
            }
            // Scheda e storico arrivano insieme da `GET /crm/clients/:id`: e'
            // l'unica rotta che uno STAFF puo' usare per vedere il passato di
            // un cliente, e gli importi li nasconde il server.
            let detail = try await crm.clientDetail(found.clientId)
            client = detail.client
            history = detail.history.filter { $0.status == .completed && $0.id != appointmentId }
        }
    }

    func markCompleted() {
        Task {
            actionError = nil
            switch await booking.markCompleted(appointmentId) {
            case .success: await load()
            case .failure(let error): actionError = error.displayMessage
            }
        }
    }

    /// "Non si è presentato": il cliente riceve una notifica; `markCompleted` lo corregge.
    func markNoShow() {
        Task {
            actionError = nil
            switch await booking.markNoShow(appointmentId) {
            case .success: await load()
            case .failure(let error): actionError = error.displayMessage
            }
        }
    }

    /// Annullato da dietro la poltrona: l'attore e' il salone, non il cliente.
    func cancel(onDone: @escaping () -> Void) {
        Task {
            actionError = nil
            switch await booking.cancel(appointmentId, by: .salon) {
            case .success: onDone()
            case .failure(let error): actionError = error.displayMessage
            }
        }
    }
}

struct AppointmentDetailScreen: View {
    @State var viewModel: AppointmentDetailViewModel
    let onBack: () -> Void

    @State private var confirmCancel = false
    @State private var confirmNoShow = false
    @State private var editSheet: StaffBookingViewModel?
    @Environment(\.container) private var container
    @Environment(\.openURL) private var openURL

    var body: some View {
        Group {
            if let apt = viewModel.appointment {
                content(apt)
            } else if viewModel.state.isLoading {
                BrandLoading().frame(maxHeight: .infinity)
            } else if let error = viewModel.state.error {
                BrandErrorRetry(error: error, retry: { Task { await viewModel.load() } })
                    .frame(maxHeight: .infinity)
            } else {
                Color.bone
            }
        }
        .background(Color.bone)
        .task { await viewModel.load() }
    }

    private func content(_ apt: Appointment) -> some View {
        let client = viewModel.client
        let completed = apt.status == .completed
        return VStack(spacing: 0) {
            DarkHeader(contentPadding: EdgeInsets(top: 0, leading: 0, bottom: 18, trailing: 0)) {
                BrandTopBar(
                    title: L("apt_detail_title_short"),
                    onBack: onBack,
                    backLabel: L("staff_tab_agenda")
                ) {
                    // Si modifica solo ciò che è ancora in programma.
                    if apt.isActive {
                        BarAction(text: L("apt_detail_edit"), action: { openEdit(apt) }, color: .oliveWood)
                    }
                }
                HStack(spacing: 12) {
                    // Iniziali oro su nero, come nelle schede del titolare; il filo
                    // chiaro stacca il riquadro dalla banda scura.
                    RoundedRectangle(cornerRadius: 16)
                        .fill(Color.ink)
                        .frame(width: 52, height: 52)
                        .overlay(
                            RoundedRectangle(cornerRadius: 16)
                                .strokeBorder(Color.bone.opacity(0.14), lineWidth: 1)
                        )
                        .overlay(
                            Text(client?.initials ?? "")
                                .font(Typo.cormorant(17, weight: .regular))
                                .foregroundStyle(Color.oliveWood)
                        )
                    VStack(alignment: .leading, spacing: 6) {
                        Text(client?.fullName ?? "")
                            .font(Typo.cormorant(24))
                            .foregroundStyle(Color.bone)
                        HStack(spacing: 7) {
                            headerPill(statusLabel(apt.status), accent: true)
                            headerPill(L("apt_detail_visits", client?.visitCount ?? 0), accent: false)
                        }
                    }
                }
                .padding(.horizontal, 20)
                .padding(.top, 6)
            }

            ScrollView {
                VStack(alignment: .leading, spacing: 14) {
                    HStack(spacing: 10) {
                        StatTile(
                            label: L("apt_detail_time"),
                            value: "\(formatTime(apt.time)) — \(formatTime(apt.end.time))"
                        )
                        // La durata al posto del totale: le cifre le vede solo il titolare.
                        StatTile(label: L("apt_detail_duration"), value: formatDuration(apt.durationMinutes))
                    }

                    VStack(alignment: .leading, spacing: 8) {
                        BrandSectionLabel(text: L("apt_detail_services"))
                        HStack(spacing: 8) {
                            ForEach(apt.serviceIds.compactMap { viewModel.services[$0] }) { service in
                                Text("\(service.name) · \(service.durationMinutes)′")
                                    .font(Typo.titleSmall)
                                    .foregroundStyle(Color.bone)
                                    .lineLimit(1)
                                    .padding(.horizontal, 14)
                                    .padding(.vertical, 11)
                                    .background(RoundedRectangle(cornerRadius: 10).fill(Color.ink))
                            }
                        }
                    }

                    // La nota scritta dal cliente nel riepilogo della prenotazione.
                    if let note = apt.noteForOperator, !note.trimmingCharacters(in: .whitespaces).isEmpty {
                        VStack(alignment: .leading, spacing: 8) {
                            BrandSectionLabel(text: L("apt_detail_client_note"))
                            Text(note)
                                .font(Typo.bodyMedium)
                                .foregroundStyle(Color.ink)
                                .frame(maxWidth: .infinity, alignment: .leading)
                                .padding(.horizontal, 15)
                                .padding(.vertical, 14)
                                .background(RoundedRectangle(cornerRadius: 16).fill(Color.stone))
                        }
                    }

                    if !viewModel.history.isEmpty {
                        VStack(alignment: .leading, spacing: 8) {
                            BrandSectionLabel(text: L("apt_detail_history"))
                            ForEach(viewModel.history) { past in
                                HStack {
                                    Text(
                                        "\(formatDateShort(past.date)) · " +
                                            past.serviceIds.compactMap { viewModel.services[$0]?.name }.joined(separator: " + ")
                                    )
                                    .font(Typo.bodyMedium)
                                    .foregroundStyle(Color.ink)
                                    .lineLimit(1)
                                    Spacer()
                                }
                                .padding(.horizontal, 15)
                                .padding(.vertical, 14)
                                .background(RoundedRectangle(cornerRadius: 16).fill(Color.stone))
                            }
                        }
                    }
                }
                .padding(.horizontal, 20)
                .padding(.top, 18)
                .padding(.bottom, 16)
                .readableWidth()
            }

            // Stato a mano: il no-show, da orario d'inizio passato.
            if apt.canMarkNoShow() {
                Button {
                    confirmNoShow = true
                } label: {
                    Text(L("apt_detail_mark_no_show"))
                        .font(Typo.titleSmall)
                        .foregroundStyle(Color.ink)
                        .padding(.horizontal, 20)
                        .padding(.vertical, 8)
                }
                .buttonStyle(.plain)
            }
            if apt.isActive {
                Button {
                    confirmCancel = true
                } label: {
                    Text(L("apt_detail_cancel"))
                        .font(Typo.titleSmall)
                        .foregroundStyle(Color.errorRed)
                        .padding(.horizontal, 20)
                        .padding(.vertical, 10)
                }
                .buttonStyle(.plain)
            }
            if let actionError = viewModel.actionError {
                Text(actionError)
                    .font(Typo.bodySmall)
                    .foregroundStyle(Color.errorRed)
                    .padding(.horizontal, 20)
                    .padding(.bottom, 6)
            }
            HStack(spacing: 10) {
                Button {
                    // Su un iPad senza telefonia openURL non fa nulla.
                    if let url = phoneDialURL(viewModel.client?.phone) { openURL(url) }
                } label: {
                    Text(L("apt_detail_call"))
                        .font(Typo.titleMedium)
                        .foregroundStyle(Color.ink)
                        .frame(maxWidth: .infinity)
                        .frame(height: 54)
                        .background(RoundedRectangle(cornerRadius: 16).fill(Color.stone))
                }
                .buttonStyle(.plain)
                // Lo stato si chiude da solo a fine servizio (§6.2): il
                // pulsante resta solo dove serve davvero, cioè per rimettere a
                // posto un no-show segnato per sbaglio.
                if apt.canRevertNoShow {
                    AccentButton(
                        text: L("apt_detail_complete"),
                        action: viewModel.markCompleted,
                        height: 54,
                        corner: 16
                    )
                    .frame(maxWidth: .infinity)
                }
            }
            .padding(.horizontal, 20)
            .padding(.bottom, 14)
            .readableWidth()
        }
        .background(Color.bone)
        .alert(L("apt_detail_cancel_title"), isPresented: $confirmCancel) {
            Button(L("apt_detail_cancel_yes"), role: .destructive) {
                viewModel.cancel(onDone: onBack)
            }
            Button(L("apt_detail_cancel_no"), role: .cancel) {}
        } message: {
            Text(L("apt_detail_cancel_body"))
        }
        .alert(L("apt_detail_no_show_title"), isPresented: $confirmNoShow) {
            Button(L("apt_detail_no_show_yes"), role: .destructive) { viewModel.markNoShow() }
            Button(L("apt_detail_cancel_no"), role: .cancel) {}
        } message: {
            Text(L("apt_detail_no_show_body"))
        }
        .sheet(item: $editSheet) { sheetViewModel in
            StaffBookingScreen(
                viewModel: sheetViewModel,
                onDismiss: {
                    let saved = sheetViewModel.done
                    editSheet = nil
                    // Salvata la modifica, questo appuntamento è stato sostituito:
                    // si torna all'agenda, dove c'è quello nuovo.
                    if saved { onBack() }
                }
            )
        }
    }

    private func openEdit(_ appointment: Appointment) {
        editSheet = StaffBookingViewModel(
            auth: container.auth, booking: container.booking,
            catalog: container.catalog, crm: container.crm,
            editAppointment: appointment, editClient: viewModel.client
        )
    }

    private func statusLabel(_ status: AppointmentStatus) -> String {
        switch status {
        case .inProgress: L("staff_in_progress")
        case .completed: L("apt_detail_completed")
        case .noShow: L("apt_detail_no_show")
        default: L("apts_status_confirmed_staff")
        }
    }

    private func headerPill(_ text: String, accent: Bool) -> some View {
        Text(text.uppercased())
            .font(Typo.jost(9, weight: .medium))
            .kerning(1.3)
            .foregroundStyle(Color.bone)
            .padding(.horizontal, 10)
            .padding(.vertical, 6)
            .background(
                RoundedRectangle(cornerRadius: 6)
                    .fill(accent ? Color.oliveWood : Color.bone.opacity(0.12))
            )
    }
}

File: android/feature/staff/src/main/kotlin/com/devora/mencare/feature/staff/AppointmentDetailScreen.kt
package com.devora.mencare.feature.staff

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.navigationBarsPadding
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.AlertDialog
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.material3.TextButton
import androidx.compose.runtime.Composable
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.saveable.rememberSaveable
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.res.stringResource
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.em
import androidx.compose.ui.unit.sp
import androidx.hilt.navigation.compose.hiltViewModel
import androidx.lifecycle.compose.collectAsStateWithLifecycle
import com.devora.mencare.core.common.formatDateShort
import com.devora.mencare.core.common.formatDuration
import com.devora.mencare.core.common.formatTime
import com.devora.mencare.core.designsystem.component.AccentButton
import com.devora.mencare.core.designsystem.component.BarAction
import com.devora.mencare.core.designsystem.component.BrandSectionLabel
import com.devora.mencare.core.designsystem.component.BrandTopBar
import com.devora.mencare.core.designsystem.component.DarkHeader
import com.devora.mencare.core.designsystem.component.StatTile
import com.devora.mencare.core.designsystem.component.readableWidth
import com.devora.mencare.core.designsystem.theme.Bone
import com.devora.mencare.core.designsystem.theme.Cormorant
import com.devora.mencare.core.designsystem.theme.ErrorRed
import com.devora.mencare.core.designsystem.theme.Ink
import com.devora.mencare.core.designsystem.theme.OliveLight
import com.devora.mencare.core.designsystem.theme.OliveWood
import com.devora.mencare.core.designsystem.theme.Stone
import com.devora.mencare.core.designsystem.util.dialPhone
import com.devora.mencare.core.model.AppointmentStatus

@Composable
fun AppointmentDetailScreen(
    onBack: () -> Unit,
    viewModel: AppointmentDetailViewModel = hiltViewModel(),
) {
    val state by viewModel.state.collectAsStateWithLifecycle()
    val apt = state.appointment ?: return
    val client = state.client
    val completed = apt.status == AppointmentStatus.COMPLETED
    var confirmCancel by rememberSaveable { mutableStateOf(false) }
    var confirmNoShow by rememberSaveable { mutableStateOf(false) }
    var editOpen by rememberSaveable { mutableStateOf(false) }
    var edited by rememberSaveable { mutableStateOf(false) }
    val context = LocalContext.current

    // Salvata la modifica, questo appuntamento è stato sostituito (e annullato):
    // si torna all'agenda, dove c'è quello nuovo.
    LaunchedEffect(apt.status, edited) {
        if (edited && apt.status == AppointmentStatus.CANCELLED) onBack()
    }

    Column(modifier = Modifier.fillMaxSize().background(Bone)) {
        DarkHeader(
            contentPadding = PaddingValues(start = 0.dp, end = 0.dp, top = 0.dp, bottom = 18.dp),
        ) {
            BrandTopBar(
                title = stringResource(R.string.apt_detail_title_short),
                onBack = onBack,
                backLabel = stringResource(R.string.staff_tab_agenda),
                // Si modifica solo ciò che è ancora in programma.
                trailing = {
                    if (apt.isActive) {
                        BarAction(
                            stringResource(R.string.apt_detail_edit),
                            {
                                edited = true
                                editOpen = true
                            },
                            color = OliveWood,
                        )
                    }
                },
            )
            Row(
                modifier = Modifier.padding(horizontal = 20.dp).padding(top = 6.dp),
                verticalAlignment = Alignment.CenterVertically,
            ) {
                // Iniziali oro su nero, come nelle schede del titolare; il filo
                // chiaro stacca il riquadro dalla banda scura.
                Box(
                    modifier = Modifier
                        .size(52.dp)
                        .clip(RoundedCornerShape(16.dp))
                        .background(Ink)
                        .border(1.dp, Bone.copy(alpha = 0.14f), RoundedCornerShape(16.dp)),
                    contentAlignment = Alignment.Center,
                ) {
                    Text(
                        listOfNotNull(client?.firstName?.firstOrNull(), client?.lastName?.firstOrNull())
                            .joinToString(""),
                        fontFamily = Cormorant,
                        fontSize = 17.sp,
                        // Oro, non oliva: sul nero l'oliva si ferma a 3.7:1.
                        color = OliveLight,
                    )
                }
                Column(Modifier.weight(1f).padding(start = 12.dp)) {
                    Text(
                        listOfNotNull(client?.firstName, client?.lastName).joinToString(" "),
                        style = MaterialTheme.typography.headlineMedium.copy(fontSize = 24.sp),
                        color = Bone,
                    )
                    Spacer(Modifier.height(6.dp))
                    Row(horizontalArrangement = Arrangement.spacedBy(7.dp)) {
                        HeaderPill(
                            text = stringResource(
                                when (apt.status) {
                                    AppointmentStatus.IN_PROGRESS -> R.string.staff_in_progress
                                    AppointmentStatus.COMPLETED -> R.string.apt_detail_completed
                                    AppointmentStatus.NO_SHOW -> R.string.apt_detail_no_show
                                    else -> R.string.apts_status_confirmed_staff
                                },
                            ),
                            accent = true,
                        )
                        HeaderPill(
                            text = stringResource(R.string.apt_detail_visits, client?.visitCount ?: 0),
                            accent = false,
                        )
                    }
                }
            }
        }

        Column(
            modifier = Modifier
                .weight(1f)
                .verticalScroll(rememberScrollState())
                .readableWidth()
                .padding(horizontal = 20.dp)
                .padding(top = 18.dp, bottom = 16.dp),
            verticalArrangement = Arrangement.spacedBy(14.dp),
        ) {
            Row(horizontalArrangement = Arrangement.spacedBy(10.dp)) {
                StatTile(
                    label = stringResource(R.string.apt_detail_time),
                    value = "${formatTime(apt.time)} — ${formatTime(apt.end.toLocalTime())}",
                    modifier = Modifier.weight(1f),
                )
                // La durata al posto del totale: le cifre le vede solo il titolare.
                StatTile(
                    label = stringResource(R.string.apt_detail_duration),
                    value = formatDuration(apt.durationMinutes),
                    modifier = Modifier.weight(1f),
                )
            }

            Column {
                BrandSectionLabel(stringResource(R.string.apt_detail_services))
                Spacer(Modifier.height(8.dp))
                Row(horizontalArrangement = Arrangement.spacedBy(8.dp)) {
                    apt.serviceIds.mapNotNull { state.services[it] }.forEach { service ->
                        Text(
                            "${service.name} · ${service.durationMinutes}'",
                            style = MaterialTheme.typography.titleSmall,
                            color = Bone,
                            maxLines = 1,
                            overflow = TextOverflow.Ellipsis,
                            modifier = Modifier
                                .weight(1f, fill = false)
                                .clip(RoundedCornerShape(10.dp))
                                .background(Ink)
                                .padding(horizontal = 14.dp, vertical = 11.dp),
                        )
                    }
                }
            }

            // La nota scritta dal cliente nel riepilogo della prenotazione.
            apt.noteForOperator?.takeIf { it.isNotBlank() }?.let { note ->
                Column {
                    BrandSectionLabel(stringResource(R.string.apt_detail_client_note))
                    Spacer(Modifier.height(8.dp))
                    Text(
                        note,
                        style = MaterialTheme.typography.bodyMedium,
                        color = Ink,
                        modifier = Modifier
                            .fillMaxWidth()
                            .clip(RoundedCornerShape(16.dp))
                            .background(Stone)
                            .padding(horizontal = 15.dp, vertical = 14.dp),
                    )
                }
            }

            if (state.history.isNotEmpty()) {
                Column {
                    BrandSectionLabel(stringResource(R.string.apt_detail_history))
                    Spacer(Modifier.height(8.dp))
                    state.history.forEach { past ->
                        Row(
                            modifier = Modifier
                                .fillMaxWidth()
                                .padding(bottom = 8.dp)
                                .clip(RoundedCornerShape(16.dp))
                                .background(Stone)
                                .padding(horizontal = 15.dp, vertical = 14.dp),
                            verticalAlignment = Alignment.CenterVertically,
                        ) {
                            Text(
                                "${formatDateShort(past.date)} · " +
                                    past.serviceIds.mapNotNull { state.services[it]?.name }.joinToString(" + "),
                                style = MaterialTheme.typography.bodyMedium,
                                color = Ink,
                                maxLines = 1,
                                overflow = TextOverflow.Ellipsis,
                                modifier = Modifier.weight(1f),
                            )
                        }
                    }
                }
            }
        }

        // Stato a mano: il no-show, da orario d'inizio passato.
        if (apt.canMarkNoShow()) {
            Text(
                stringResource(R.string.apt_detail_mark_no_show),
                style = MaterialTheme.typography.titleSmall,
                color = Ink,
                modifier = Modifier
                    .align(Alignment.CenterHorizontally)
                    .clickable { confirmNoShow = true }
                    .padding(horizontal = 20.dp, vertical = 8.dp),
            )
        }
        if (apt.isActive) {
            Text(
                stringResource(R.string.apt_detail_cancel),
                style = MaterialTheme.typography.titleSmall,
                color = ErrorRed,
                modifier = Modifier
                    .align(Alignment.CenterHorizontally)
                    .clickable { confirmCancel = true }
                    .padding(horizontal = 20.dp, vertical = 10.dp),
            )
        }
        Row(
            modifier = Modifier
                .navigationBarsPadding()
                .readableWidth()
                .padding(horizontal = 20.dp)
                .padding(bottom = 14.dp),
            horizontalArrangement = Arrangement.spacedBy(10.dp),
        ) {
            Box(
                modifier = Modifier
                    .weight(1f)
                    .height(54.dp)
                    .clip(RoundedCornerShape(16.dp))
                    .background(Stone)
                    .clickable(enabled = client?.phone?.isNotBlank() == true) { context.dialPhone(client?.phone) },
                contentAlignment = Alignment.Center,
            ) {
                Text(
                    stringResource(R.string.apt_detail_call),
                    style = MaterialTheme.typography.titleMedium,
                    color = Ink,
                )
            }
            // Lo stato si chiude da solo a fine servizio (§6.2): il pulsante
            // resta solo dove serve davvero, cioè per rimettere a posto un
            // no-show segnato per sbaglio.
            if (apt.canRevertNoShow) {
                AccentButton(
                    text = stringResource(R.string.apt_detail_complete),
                    onClick = viewModel::markCompleted,
                    height = 54.dp,
                    shape = RoundedCornerShape(16.dp),
                    modifier = Modifier.weight(1.2f),
                )
            }
        }
    }

    if (confirmNoShow) {
        AlertDialog(
            onDismissRequest = { confirmNoShow = false },
            containerColor = Bone,
            title = { Text(stringResource(R.string.apt_detail_no_show_title)) },
            text = { Text(stringResource(R.string.apt_detail_no_show_body)) },
            confirmButton = {
                TextButton(onClick = {
                    confirmNoShow = false
                    viewModel.markNoShow()
                }) {
                    Text(stringResource(R.string.apt_detail_no_show_yes), color = ErrorRed)
                }
            },
            dismissButton = {
                TextButton(onClick = { confirmNoShow = false }) {
                    Text(stringResource(R.string.apt_detail_cancel_no), color = Ink)
                }
            },
        )
    }

    if (editOpen) {
        StaffBookingScreen(
            onDismiss = { editOpen = false },
            editAppointment = apt,
            editClient = client,
            sessionKey = "staff-edit-${apt.id}",
        )
    }

    if (confirmCancel) {
        AlertDialog(
            onDismissRequest = { confirmCancel = false },
            containerColor = Bone,
            title = { Text(stringResource(R.string.apt_detail_cancel_title)) },
            text = { Text(stringResource(R.string.apt_detail_cancel_body)) },
            confirmButton = {
                TextButton(onClick = {
                    confirmCancel = false
                    // L'annullo torna indietro da sé: non deve scattare anche il ritorno della modifica.
                    edited = false
                    viewModel.cancel(onBack)
                }) {
                    Text(stringResource(R.string.apt_detail_cancel_yes), color = ErrorRed)
                }
            },
            dismissButton = {
                TextButton(onClick = { confirmCancel = false }) {
                    Text(stringResource(R.string.apt_detail_cancel_no), color = Ink)
                }
            },
        )
    }
}

@Composable
private fun HeaderPill(text: String, accent: Boolean) {
    Text(
        text.uppercase(),
        style = MaterialTheme.typography.labelSmall.copy(fontSize = 9.sp, letterSpacing = 0.14.em),
        color = Bone,
        modifier = Modifier
            .clip(RoundedCornerShape(6.dp))
            .background(if (accent) OliveWood else Bone.copy(alpha = 0.12f))
            .padding(horizontal = 10.dp, vertical = 6.dp),
    )
}
